client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java

```diff
@@ -246,32 +247,74 @@ public <K, V> ShuffleWriter<K, V> getWriter(
         int shuffleId = SparkUtils.celebornShuffleId(shuffleClient, h, context, true);
         shuffleIdTracker.track(h.shuffleId(), shuffleId);
 
-        if (ShuffleMode.SORT.equals(celebornConf.shuffleWriterMode())) {
-          ExecutorService pushThread =
-              celebornConf.clientPushSortPipelineEnabled() ? getPusherThread() : null;
-          return new SortBasedShuffleWriter<>(
-              shuffleId,
-              h.dependency(),
-              h.numMappers(),
-              context,
-              celebornConf,
-              shuffleClient,
-              metrics,
-              pushThread,
-              SendBufferPool.get(cores, sendBufferPoolCheckInterval, sendBufferPoolExpireTimeout));
-        } else if (ShuffleMode.HASH.equals(celebornConf.shuffleWriterMode())) {
-          SendBufferPool pool =
-              SendBufferPool.get(cores, sendBufferPoolCheckInterval, sendBufferPoolExpireTimeout);
-          if (COLUMNAR_SHUFFLE_CLASSES_PRESENT && celebornConf.columnarShuffleEnabled()) {
-            return SparkUtils.createColumnarHashBasedShuffleWriter(
-                shuffleId, h, context, celebornConf, shuffleClient, metrics, pool);
+        if (celebornConf.shuffleWriteModeByPartitionCountEnable()) {
+          int partitionCount = h.dependency().partitioner().numPartitions();
+          if (partitionCount > celebornConf.shuffleWriteModeByPartitionCountThreshold()) {
```

Comment:
The code seems a bit repetitive. How about using the `shuffleMode` variable to dynamically switch modes?

```java
        ShuffleMode shuffleMode = celebornConf.shuffleWriterMode();
        if (celebornConf.shuffleWriteModeByPartitionCountEnable()) {
          int partitionCount = h.dependency().partitioner().numPartitions();
          if (partitionCount > celebornConf.shuffleWriteModeByPartitionCountThreshold()) {
            shuffleMode = ShuffleMode.SORT;
            logger.info(
                "Shuffle {} write mode is changed to {} because  "
                    + "partition count {} is greater than threshold {}",
                shuffleId,
                shuffleMode,
                partitionCount,
                celebornConf.shuffleWriteModeByPartitionCountThreshold());
          } else {
            shuffleMode = ShuffleMode.HASH;
            logger.info(
                "Shuffle {} write mode is changed to {} because "
                    + "partition count {} is less than threshold {}",
                shuffleId,
                shuffleMode,
                partitionCount,
                celebornConf.shuffleWriteModeByPartitionCountThreshold());
          }
        }
```

Reply:
Thank you for your suggestion.I have updated